common/djangoapps/util/views.py

```diff
@@ -187,3 +193,33 @@ def inner(request, *args, **kwargs):
         response['P3P'] = settings.P3P_HEADER
         return response
     return inner
+
+
+@ensure_csrf_cookie
+def reset_course_deadlines(request):
+    """
+    Set the start_date of a schedule to today, which in turn will adjust due dates for
+    sequentials belonging to a self paced course
+    """
+    from lms.urls import RENDER_XBLOCK_NAME
+    from openedx.features.course_experience.urls import COURSE_HOME_VIEW_NAME
+
+    detail_id_dict = ast.literal_eval(request.POST.get('reset_deadlines_redirect_url_id_dict'))
```

Comment:
I think it is personal preference, but you could also use json.loads to perform this convert to a dict (if you also do a replace from ' to " (or vice versa) first.)